db/migrate/20230925123456_add_design_file_id_to_text_layers.rb

```diff
@@ -0,0 +1,7 @@
+class AddDesignFileIdToTextLayers < ActiveRecord::Migration[7.0]
+  def change
+    unless column_exists? :text_layers, :design_file_id
+      add_reference :text_layers, :design_file, null: false, foreign_key: true
+    end
+  end
+end
```

Comment:
The migration file for adding a design_file_id to text_layers does not specify an index for the foreign key. This can lead to performance issues when querying the text_layers table. Consider adding an index with 'add_index :text_layers, :design_file_id'.